File: scratch/patch_form.py

import os

filepath = r'c:\2035-HMS\SAAS_ERP\src\app\settings\hms\hms-settings-form.tsx'
with open(filepath, 'r', encoding='utf-8') as f:
    lines = f.readlines()

new_lines = []
found = False
for line in lines:
    new_lines.append(line)
    if "router.refresh();" in line and not found:
        # Check if we are inside onSaveTemplate
        # (Very naive check but let's hope)
        # Looking at context around lines 864
        new_lines.append(line.replace("router.refresh();", "return res;").replace("router.refresh();", "")) # This is wrong logic

# Let's try again with actual matching
content = "".join(lines)
old_block = """                                                    toast({ title: "Template Saved", description: `'${name}' updated.` });
                                                    router.refresh();
                                                }"""

# I will use a more robust search
target = 'toast({ title: "Template Saved", description: `\'${name}\' updated.` });'
if target in content:
    print("Found target!")
    new_content = content.replace('router.refresh();\n                                                }', 'router.refresh();\n                                                }\n                                                return res;')
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(new_content)
    print("Replacment done!")
else:
    print("Target NOT found!")

File: scratch/patch_form.js

const fs = require('fs');
const path = 'c:\\2035-HMS\\SAAS_ERP\\src\\app\\settings\\hms\\hms-settings-form.tsx';
let content = fs.readFileSync(path, 'utf8');

if (content.includes('toast({ title: "Template Saved"')) {
    console.log("Found target!");
    // Look for the router.refresh(); line inside onSaveTemplate
    // and add return res; after it.
    const searchString = 'router.refresh();\n                                                }';
    if (content.includes(searchString)) {
        console.log("Found block with specific indentation!");
        content = content.replace(searchString, 'router.refresh();\n                                                }\n                                                return res;');
        fs.writeFileSync(path, content);
        console.log("Applied replacement!");
    } else {
        console.log("Could NOT find specific block. Trying regex...");
        content = content.replace(/(router\.refresh\(\);[\s\n]+})/g, '$1\n                                                return res;');
        fs.writeFileSync(path, content);
        console.log("Applied regex replacement!");
    }
} else {
    console.log("Target NOT found at all!");
}
